### DISCUSSION TITLE
Why the mixin helper isn't working for source only library?

### DISCUSSION THREAD

**Discussion by SettingDust** [minecraft-dev/MinecraftDev#2334] [Q&A] [answered]
I'm on 1.12.2. Using RetroGradle. The mixin target Minecraft all error in Minecraft Dev. But works in game

![图片](https://github.com/user-attachments/assets/99e7b789-44c0-4d5e-9610-dd4161db89ed)

**Comment by Earthcomputer** [comment, marked as answer]
Hi, Mixin works with bytecode and therefore mcdev's mixin features don't work if there is no bytecode available.
